drivers/src/main/java/org/gorpipe/oci/driver/OCIUrl.java

```diff
@@ -156,8 +154,8 @@ private static OCIUrl parseS3HttpUrl(URI uri) throws MalformedURLException {
     }
 
     private static OCIUrl parseGlobalUrl(URI uri) throws MalformedURLException {
```

Comment:
See that we are using the URI object. Wasnt there a performance hit using it?

Reply:
It was in the dictionaries when you 1M uris to parse, and you are only using small portion to actually open files.  Here, compared to open the file the overhead should be acceptable.

Reply:
Plus, here we need full parsing of the urls (as we are extracting so much info from it), which we did not really need in the dicts.